packages/types/src/credentials.ts

```diff
@@ -1,31 +1,14 @@
+import { AwsCredentialIdentity } from "./identity";
 import { Provider } from "./util";
 
-export interface AwsCredentialIdentity {
-  /**
-   * A {Date} when the identity or credential will no longer be accepted.
-   */
-  readonly expiration?: Date;
-}
-
 /**
  * An object representing temporary or permanent AWS credentials.
+ *
+ * @deprecated Use {@AwsCredentialIdentity}
  */
-export interface Credentials extends AwsCredentialIdentity {
-  /**
-   * AWS access key ID
-   */
-  readonly accessKeyId: string;
-
-  /**
-   * AWS secret access key
-   */
-  readonly secretAccessKey: string;
-
-  /**
-   * A security or session token to use with these credentials. Usually
-   * present for temporary credentials.
-   */
-  readonly sessionToken?: string;
-}
+export interface Credentials extends AwsCredentialIdentity {}
 
+/**
+ * @deprecated Use {@AwsCredentialIdentityProvider}
+ */
 export type CredentialProvider = Provider<Credentials>;
```

Comment:
Should this be the following?

```suggestion
/**
 * @deprecated Use {@AwsCredentialIdentityProvider}
 */
export type CredentialProvider = IdentityProvider<Credentials>;
```